kedro/extras/datasets/tensorflow/tensorflow_model_data_set.py

```diff
@@ -0,0 +1,166 @@
+        >>> import tensorflow as tf
+        >>> import numpy as np
+        >>> data_set = TensorFlowModelDataset("saved_model_path")
+        >>> model = tf.keras.Model()
+        >>> predictions = model.predict([...])
+        >>> data_set.save(model)
+        >>> loaded_model = data_set.load()
+        >>> new_predictions = loaded_model.predict([...])
+        >>> np.testing.assert_allclose(predictions, new_predictions, rtol=1e-6, atol=1e-6)
+
+    """
+
+    DEFAULT_LOAD_ARGS = dict(custom_objects=None, compile=True,)  # type: Dict[str, Any]
+
+    DEFAULT_SAVE_ARGS = dict(
+        overwrite=True,
+        include_optimizer=True,
+        save_format="tf",
+        signatures=None,
+        options=None,
+    )  # type: Dict[str, Any]
+
+    def __init__(
+            self,
+            filepath: str,
+            load_args: Dict[str, Any] = None,
+            save_args: Dict[str, Any] = None,
+            version: Version = None,
+            credentials: Dict[str, Any] = None,
+            fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """
+        Args:
+            filepath: Filepath to a TensorFlow model directory prefixed with a protocol like `s3://`.
+                If prefix is not provided `file` protocol (local filesystem) will be used.
+                The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            load_args: TensorFlow options for loading models.
+                custom_objects: Optional dictionary mapping names (strings) to custom classes or functions to be
+                considered during deserialization.
+                compile: Boolean, whether to compile the model after loading.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/load_model
+                All defaults are preserved.
+            save_args: TensorFlow options for saving models.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/save_model
+                All defaults are preserved, except for "index", which is set to False.
+                In TF this defaults to 'tf' in TF 2.X, and 'h5' in TF 1.X.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{'token': None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class.
+                E.g. for ``GCSFileSystem`` class: `{project: 'my-project', ...}`
+        """
+        _fs_args = copy.deepcopy(fs_args) or {}
```

Comment:
```suggestion
        _fs_args = copy.deepcopy(fs_args) or {}
        _fs_open_args_load = _fs_args.pop("open_args_load", {})
        _fs_open_args_save = _fs_args.pop("open_args_save", {})
```

Sorry for changing this out under you, but this is the new format we're exposing for letting users specify load and save args. :)